### PR TITLE
Add support for EIP-1559 smart contract transactions.

### PR DESCRIPTION
**Pull request by alexdupre**
### What does this PR do?
It allows to send EIP-1559 transactions from smart contract wrappers.

### Where should the reviewer start?
Instantiate the `StaticEIP1559GasProvider` and pass it when loading the contract wrapper.

### Why is it needed?
https://github.com/web3j/web3j/issues/1517

### CODE REVIEW

**Comment by alexdupre** [comment]
May I know what's the plan @andrii-kl , since you assigned it to yourself but tagged it as `Don't merge`?

**Review by gtebrean — COMMENTED** [review]
Please add a test case for your changes. You can take [this](https://github.com/web3j/web3j/blob/6f8b8b8a128c76b4a868cc544a591bdf72f586d5/core/src/test/java/org/web3j/tx/ContractTest.java#L454) as example.

File: core/src/main/java/org/web3j/tx/gas/ContractEIP1559GasProvider.java
Comment (alexdupre) [reply core/src/main/java/org/web3j/tx/gas/ContractEIP1559GasProvider.java:18]: The reason to have the method is that the underlying blockchain may not have activated the EIP-1559 extension yet. In this way the gas provider can get the information from the blockchain (or in any other way) and transactions will start to use the new type only when activated.

**Comment by alexdupre** [comment]
> Please add a test case for your changes. You can take [this](https://github.com/web3j/web3j/blob/6f8b8b8a128c76b4a868cc544a591bdf72f586d5/core/src/test/java/org/web3j/tx/ContractTest.java#L454) as example.

Test added.

**Comment by michaelmoebryan** [comment]
Hi I am looking to implement EIP1559 style contract transactions using web3j. What more needs to be done on this PR before it can be merged?

**Comment by alexdupre** [comment]
No idea, I'm using my patches in production without issues. If you like to build new jars this branch aggregates all my open pull requests: https://github.com/alexdupre/web3j/tree/local

File: core/src/main/java/org/web3j/tx/gas/StaticEIP1559GasProvider.java
Comment (andrii-kl) [inline core/src/main/java/org/web3j/tx/gas/StaticEIP1559GasProvider.java:21]: We can make all variables  final

**Review by andrii-kl — APPROVED** [review]

**Pull request merged**
